lib/api/src/lib.rs

```diff
@@ -370,5 +370,18 @@ pub use wasmer_engine_universal::{Universal, UniversalArtifact, UniversalEngine}
 #[cfg(feature = "dylib")]
 pub use wasmer_engine_dylib::{Dylib, DylibArtifact, DylibEngine};
 
+#[cfg(any(feature = "jit", feature = "native"))]
+compile_error!(
+    r#"At least one deprecated feature has been used, and cannot not
+be used any longer. Please refer to the following table:
+
+| Deprecated feature | New feature to use instead |
+|--------------------|----------------------------|
+| `jit`              | `universal`                |
+| `native`           | `dylib`                    |
+
+"#
+);
+
```

Comment:
We don't need to make of this a breaking change (making people to update their code mandatory).

We can actually do it in a better way. How?
Since we are just renaming we can do:

```rust
#[cfg(feature = "jit")]
#[deprecated(
    since = "1.1.0",
    note = "Please use the `universal` feature instead"
)]
pub type JIT = Universal;

#[cfg(feature = "native")]
#[deprecated(
    since = "1.1.0",
    note = "Please use the `native` feature instead"
)]
pub type Native = Dylib;

```

Reply:
It's more than that, it's also about the `*Artifact` type, the `*Engine` type, the `*EngineInner` type etc.